### ISSUE TITLE
Version 4.1.1 doesn't run, gives white screen on Ubuntu

### ISSUE THREAD

**Issue by philh7456** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#2103]
I previously had 4.0.5 running fine on Ubuntu 16:10, 64bit. Upgraded to 4.1.1 and GPMDP either does not start or starts but just gives a white screen with no content. I have tried deleting the config files and starting again, but no luck.
Starting from the command line shows the following:

Starting Sentry
error: Failed to load bonjour with error: {}
Bonjour is required to use Chromecast Support or to enable ZeroConf for the PlaybackAPI
On linux you need to install "avahi"
Failed to encode dictionary
Failed to encode reply value
Failed to encode dictionary
Failed to encode reply value

avahi-daemon is installed and up-to-date (0.6.32-1ubuntu1).

I have tried generating a debug file, but pressing Ctrl-shift-g in the white window does nothing.

Downgraded to 4.05, works fine.

**Comment by MarshallOfSound** [comment]
Looks like a transitive dependency bump broke something.  I would pull the linux update but I can't as it isn't technically an update (people download manually).  I've pulled the bintray release though.

/cc @jostrander can you investigate?

**Comment by oleerik** [comment]
Same issue on fedora25 :cactus:

**Comment by MarshallOfSound** [comment]
I deleted the uploaded `4.1.1` linux artifacts so no one else will download into this issue.

**Comment by voria** [comment]
I've just built the project from the current master branch (c3081d2) on Fedora 25 x86_64, starting from a clean tree. The resulting rpm from the command `npm install && npm run make:rpm:64` works perfectly fine, no issues whatsoever.

**Comment by MarshallOfSound** [comment]
@voria To confirm, building locally for you works as expected but the previously published builds did not?

**Comment by voria** [comment]
@MarshallOfSound I had no chance to try the published builds, they were already taken off when I noticed the new 4.1.1 release. If they are still available for download somewhere, I can test them.

**Comment by MarshallOfSound** [comment]
Try this one out for size 👍 

https://1489-40008106-gh.circle-artifacts.com/3/home/ubuntu/Google-Play-Music-Desktop-Player-UNOFFICIAL-/dist/installers/redhat/google-play-music-desktop-player-4.1.1.x86_64.rpm

**Comment by voria** [comment]
It does NOT work, exactly as reported. At start, GPMDP gives a white screen with no content. This is the terminal output (nothing relevant I think):

> Starting Sentry
> *** WARNING *** The program 'Google Play Music Desktop Player' uses the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=Google Play Music Desktop Player>
> *** WARNING *** The program 'Google Play Music Desktop Player' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=Google Play Music Desktop Player&f=DNSServiceRegister>

The locally built rpm works flawlessly and prints out a similar log in the terminal:

> error: Last.fm API keys have not been set!
> *** WARNING *** The program 'Google Play Music Desktop Player' uses the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=Google Play Music Desktop Player>
> *** WARNING *** The program 'Google Play Music Desktop Player' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=Google Play Music Desktop Player&f=DNSServiceRegister>
> *** WARNING *** The program 'Google Play Music Desktop Player' uses the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=Google Play Music Desktop Player>
> *** WARNING *** The program 'Google Play Music Desktop Player' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
> *** WARNING *** Please fix your application to use the native API of Avahi!
> *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=Google Play Music Desktop Player&f=DNSServiceRegister>

**Comment by MarshallOfSound** [comment]
Strange, can you run the following

```bash
git checkout ceb20222a9965da6bd711b1eaafd95561430d260
rm -rf node_modules
npm install
npm run make:rpm:64
```

And try that build.  (That commit is the `v4.1.1` hash).  Just want to check if I "fixed" it by accident in a recent commit or if the build agents made broken builds.

**Comment by voria** [comment]
The locally built rpm from `v4.1.1` hash also works flawlessly (same log output in the terminal as before).
Definitely, the build agents made broken builds.

**Comment by MarshallOfSound** [comment]
Hmmm, gonna fix the linting errors and fire off some new builds

**Comment by voria** [comment]
If you need any help for testing new builds or anything else, let me know.

**Comment by BrianAllred** [comment]
Interesting, I also tried to build at the 4.1.1 hash and got the same white screen problem. Trying again.

**Comment by voria** [comment]
@MarshallOfSound I've just tested the artifact built on [circleci](https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/1510) from latest commit (c780bf5) and it still gives the white screen problem. Instead, the resulting rpm built locally on my machine from the same commit works as expected. There must be something wrong in build agents configuration, or maybe some missing/mismatch dependency? Doing a diff between circleci build logs and local build logs, I cannot find any relevant difference though.

**Comment by MarshallOfSound** [comment]
@voria Must be a native dependency issue, I'll try reconfiguring the build agent from scratch when I get time.  But I'll build the next release in a linux VM 👍

**Comment by voria** [comment]
@MarshallOfSound I have some news. The rpm built locally from the current latest commit (d2bfd07) does NOT work, it gives the white screen problem as described previously.
So, I did some further testing, and I found that the problem started with commit 67567c4. The rpm built from its preceding commit 6a016f2 works as expected instead.

**Comment by MarshallOfSound** [comment]
@voria Can you upgrade Electron to 1.5.0 on your local on latest master and see if that had the same issue?

**Comment by voria** [comment]
@MarshallOfSound Bumping electron to 1.5.0 in _packages.json_ makes no difference. Still white screen.

**Comment by voria** [comment]
@MarshallOfSound I did some more testing and found the cause of the problem. The culprit is commit 67567c4, everything works flawlessly after reverting it.

To be more precise, look at file `src/renderer/ui/components/generic/PlatformSpecific.js`. At line 5, `semver.parse()` returns `null` on my system: apparently it does not like the string `4.9.5-200.fc25.x86_64` returned by `os.release()`. Then, at line 6, there is an uncaught error caused from the attempt to read the properties of the `null` object.

**Comment by jostrander** [comment]
Looks like there are a lot of reasons for it to return null:
https://github.com/npm/node-semver/blob/master/semver.js#L236-L255
```
function parse(version, loose) {
  if (version instanceof SemVer)
    return version;

  if (typeof version !== 'string')
    return null;

  if (version.length > MAX_LENGTH)
    return null;

  var r = loose ? re[LOOSE] : re[FULL];
  if (!r.test(version))
    return null;

  try {
    return new SemVer(version, loose);
  } catch (er) {
    return null;
  }
}
```

It seems that the `_` in the version string is causing the issue.

**Comment by MarshallOfSound** [comment]
This version prop is only used on windows currently, I'll just ignore it on linux for now

**Comment by voria** [comment]
I have some more info that may be useful, regarding the non-working published builds for `v4.1.1` that you asked me to test some days ago. I tested it again, this time by exporting the environmental variable `ELECTRON_ENABLE_LOGGING=1` in order to get some more useful log messages:

> [3501:0130/093154:INFO:CONSOLE(457)] "Uncaught Error: Cannot find module 'jquery'", source: module.js (457)
> [3501:0130/093154:INFO:CONSOLE(51)] "Uncaught ReferenceError: Logger is not defined", source: /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/rendererEmitter.js (51)

**Comment by MarshallOfSound** [comment]
> Uncaught Error: Cannot find module 'jquery'", source: module.js (457)

Weow... That's super weird
